src/gpu/nodeKernels.cu

```diff
@@ -21,12 +21,21 @@
 #include <thrust/device_ptr.h>
 #include <thrust/scan.h>
 
-__global__ void kFormat(size_t pointCount, size_t pointSize, size_t fieldCount, const GPUFieldDesc* fields, char* out)
+__global__ void kFormatSoaToAos(size_t pointCount, size_t pointSize, size_t fieldCount, const GPUFieldDesc *soaInData, char *aosOutData)
 {
 	LIMIT(pointCount);
 	// Implement padding
 	for (size_t i = 0; i < fieldCount; ++i) {
-		memcpy(out + pointSize * tid + fields[i].dstOffset, fields[i].data + fields[i].size * tid, fields[i].size);
+		memcpy(aosOutData + pointSize * tid + soaInData[i].dstOffset, soaInData[i].readDataPtr + soaInData[i].size * tid, soaInData[i].size);
+	}
+}
+
+__global__ void kFormatAosToSoa(size_t pointCount, size_t pointSize, size_t fieldCount, const char* aosInData, GPUFieldDesc* soaOutData)
+{
+	LIMIT(pointCount);
+	// Implement padding
+	for (size_t i = 0; i < fieldCount; ++i) {
+		memcpy(soaOutData[i].writeDataPtr + soaOutData[i].size * tid, aosInData + pointSize * tid + soaOutData[i].dstOffset, soaOutData[i].size);
 	}
```

Comment:
Hmm, I was boggled by this comment `// Implement padding` here and in SoaToAos. IMHO, it should work because we increase the offset for dummy (padding) fields.

Reply:
This is a comment from the previous implementation. For me either, padding is already implemented. Could some optimizations be applied? If nothing comes to your head, I will remove this comment.